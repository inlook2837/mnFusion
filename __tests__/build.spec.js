jest.setTimeout(600000)

const create = require('@vue/cli-test-utils/createTestProject')
const { defaultPreset } = require('@vue/cli/lib/options')
const path = require('path')
const fs = require('fs-extra')
const Application = require('spectron').Application

const projectPath = p => path.join(process.cwd(), '__tests__/projects/build', p)

test('build:electron', async () => {
  let project
  // Install vcp-electron-builder
  defaultPreset.plugins['vue-cli-plugin-electron-builder'] = {
    version: 'file:../../../'
  }
  project = await create(
    'build',
    defaultPreset,
    path.join(process.cwd(), '/__tests__/projects')
  )
  //   })
  const { stdout } = await project.run(
    'vue-cli-service build:electron --x64 --win zip --linux zip'
  )
  //   Ensure built completes
  expect(stdout.indexOf('Build complete!')).not.toBe(-1)
  //   Ensure /dist is not modified
  expect(project.has('dist')).toBe(false)
  //   Ensure build successfully outputted files
  expect(project.has('dist_electron/bundled/index.html')).toBe(true)
  expect(project.has('dist_electron/bundled/favicon.ico')).toBe(true)
  expect(project.has('dist_electron/bundled/js')).toBe(true)
  expect(project.has('dist_electron/bundled/css')).toBe(true)
  expect(project.has('dist_electron/bundled/background.js')).toBe(true)
  expect(project.has('dist_electron/win-unpacked/build.exe')).toBe(true)
  expect(project.has('dist_electron/linux-unpacked/build')).toBe(true)
  //   Ensure that setup file was not created
  expect(project.has('dist_electron/build Setup 0.1.0.exe')).toBe(false)
  // Ensure that zip files were created
  expect(project.has('dist_electron/build-0.1.0-win.zip')).toBe(true)
  expect(project.has('dist_electron/build-0.1.0.zip')).toBe(true)
  //   Ensure base is set properly (for app protocol)
  const index = fs.readFileSync(
    projectPath('dist_electron/bundled/index.html'),
    'utf8'
  )
  expect(index.indexOf('<base href=app://./ >')).not.toBe(-1)
  const app = new Application({
    path: './__tests__/projects/build/dist_electron/win-unpacked/build.exe'
    //   path: '../electron-test/dist_electron/win-unpacked/electron-test.exe'
  })
  await app.start()
  const win = app.browserWindow
  const client = app.client
  await client.waitUntilWindowLoaded()

  await client.getRenderProcessLogs().then(logs => {
    logs.forEach(log => {
      //   Make sure there are no fatal errors
      expect(log.level).not.toBe('SEVERE')
    })
  })
  await client.getMainProcessLogs().then(logs => {
    logs.forEach(log => {
      //   Make sure there are no fatal errors
      expect(log.level).not.toBe('SEVERE')
    })
  })
  //   Window was created
  expect(await client.getWindowCount()).toBe(1)
  //   It is not minimized
  expect(await win.isMinimized()).toBe(false)
  //   Dev tools is not open
  expect(await win.isDevToolsOpened()).toBe(false)
  //   Window is visible
  expect(await win.isVisible()).toBe(true)
  //   Size is correct
  const { width, height } = await win.getBounds()
  expect(width).toBeGreaterThan(0)
  expect(height).toBeGreaterThan(0)
  // Load was successful
  expect(await app.webContents.isLoading()).toBe(false)
  //   App is loaded properly
  expect(await client.getHTML('#app')).toMatchSnapshot()

  app.stop()
})
