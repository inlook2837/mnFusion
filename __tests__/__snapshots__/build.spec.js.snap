// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build:electron 1`] = `
"<div id=\\"app\\"><img src=\\"./img/logo.82b9c7a5.png\\"><div data-v-20cac705=\\"\\" class=\\"hello\\"><h1 data-v-20cac705=\\"\\">Welcome to Your Vue.js App</h1><p data-v-20cac705=\\"\\">
    For guide and recipes on how to configure / customize this project,<br data-v-20cac705=\\"\\">
    check out the
    <a data-v-20cac705=\\"\\" href=\\"https://cli.vuejs.org\\" target=\\"_blank\\">vue-cli documentation</a>.
  </p><h3 data-v-20cac705=\\"\\">Installed CLI Plugins</h3><ul data-v-20cac705=\\"\\"><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-babel\\" target=\\"_blank\\">babel</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://github.com/vuejs/vue-cli/tree/dev/packages/%40vue/cli-plugin-eslint\\" target=\\"_blank\\">eslint</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://www.npmjs.com/package/vue-cli-plugin-electron-builder\\" target=\\"_blank\\">electron-builder</a></li></ul><h3 data-v-20cac705=\\"\\">Essential Links</h3><ul data-v-20cac705=\\"\\"><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://vuejs.org\\" target=\\"_blank\\">Core Docs</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://forum.vuejs.org\\" target=\\"_blank\\">Forum</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://chat.vuejs.org\\" target=\\"_blank\\">Community Chat</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://twitter.com/vuejs\\" target=\\"_blank\\">Twitter</a></li></ul><h3 data-v-20cac705=\\"\\">Ecosystem</h3><ul data-v-20cac705=\\"\\"><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://router.vuejs.org\\" target=\\"_blank\\">vue-router</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://vuex.vuejs.org\\" target=\\"_blank\\">vuex</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://github.com/vuejs/vue-devtools#vue-devtools\\" target=\\"_blank\\">vue-devtools</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://vue-loader.vuejs.org\\" target=\\"_blank\\">vue-loader</a></li><li data-v-20cac705=\\"\\"><a data-v-20cac705=\\"\\" href=\\"https://github.com/vuejs/awesome-vue\\" target=\\"_blank\\">awesome-vue</a></li></ul></div></div>"
`;
